.github/workflows/ci.yml

```diff
@@ -84,28 +84,31 @@ jobs:
         name: Prepare LD_LIBRARY_PATH (Ubuntu only)
         run: echo LD_LIBRARY_PATH=${pythonLocation}/lib >> $GITHUB_ENV
 
-      - name: Build docs
-        run: cargo doc --features "num-bigint num-complex hashbrown" --verbose --target ${{ matrix.platform.rust-target }}
+      # Validate docs have no warnings (except on PyPy; auto-initialize feature fails the compile until `cfg(rustdoc)` is stable).
+      - if: matrix.python-version != 'pypy-3.6'
+        name: Build docs
+        run: cargo doc --no-default-features --features "macros num-bigint num-complex hashbrown" --verbose --target ${{ matrix.platform.rust-target }}
 
-      - name: Build without default features
+      - name: Build (no features)
         run: cargo build --no-default-features --verbose --target ${{ matrix.platform.rust-target }}
 
-      - name: Build with default features
-        run: cargo build --features "num-bigint num-complex hashbrown" --verbose --target ${{ matrix.platform.rust-target }}
+      - name: Build (all additive features)
+        run: cargo build --no-default-features --features "macros num-bigint num-complex hashbrown" --verbose --target ${{ matrix.platform.rust-target }}
 
       # Run tests (except on PyPy, because no embedding API).
       - if: matrix.python-version != 'pypy-3.6'
         name: Test
-        run: cargo test --features "num-bigint num-complex hashbrown" --target ${{ matrix.platform.rust-target }}
+        run: cargo test --no-default-features --features "macros num-bigint num-complex hashbrown" --target ${{ matrix.platform.rust-target }}
```

Comment:
Why not using `auto-initialize`?

Reply:
It's automatically included in tests because I put this in `[dev-dependencies]`:

```
# features needed to run the PyO3 test suite
pyo3 = { path = ".", default-features = false, features = ["macros", "auto-initialize"] }
```

However, that's also the reason why `auto-initialize` and `macros` are included in the `cargo build` command above. Interestingly `cargo` is about to stabilize a fix for this as a new `resolver = "2"` option in Cargo.toml. https://github.com/rust-lang/cargo/pull/8997 (I tested on nightly and the features are no longer included in `cargo build`.)

Because in the future what I've done here _will_ work correctly in the future, I'm going to keep the code mostly as-is for the moment and introduce a hack to avoid the immediate issue with a TODO to tidy it up once we bump MSRV in the future (probably to Rust 1.52).